src/pages/index.js

```diff
@@ -0,0 +1,283 @@
+  });
+
+  return cardElement;
+}
+
+// Open & Close Modal Functions
+function handleLikeButton(evt, _id) {
+  const likeButton = evt.target;
+  const isLiked = likeButton.classList.contains("card__like-button_liked");
+
+  api
+    .handleLike(_id, isLiked)
+    .then(() => {
+      likeButton.classList.toggle("card__like-button_liked");
+    })
+    .catch(console.error);
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscapeKey);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscapeKey);
+}
+
+function handleEscapeKey(evt) {
+  if (evt.key === "Escape") {
+    const openedModal = document.querySelector(".modal_opened");
+    if (openedModal) closeModal(openedModal);
+  }
+}
+
+// DELETE card
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+
+  setButtonText(submitBtn, true, "Yes", "Deleting...");
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false, "Delete", "Deleting...");
+    });
+}
+
+// PROFILE edit
+function handleProfileFormEdit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+
+  setButtonText(submitBtn, true, "Save", "Saving...");
+
+  api
+    .editUserInfo({
+      name: editModalName.value,
+      about: editModalDescription.value,
+    })
+    .then((data) => {
+      profileNameElement.textContent = data.name;
+      profileDescriptionElement.textContent = data.about;
+      closeModal(profileEditModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false, "Save", "Saving...");
+    });
+}
+
+// AVATAR update
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+
+  setButtonText(submitBtn, true, "Save", "Saving...");
+
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      profileAvatarElement.src = data.avatar;
+      closeModal(avatarModal);
+      avatarForm.reset();
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false, "Save", "Saving...");
+    });
+}
+
+// ADD Card to server
+function handleCardSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+
+  setButtonText(submitBtn, true, "Create", "Saving...");
+
+  api
+    .addCard({
+      name: cardNameInput.value,
+      link: cardLinkInput.value,
+    })
+    .then((createdCard) => {
+      const cardElement = getCardElement(createdCard);
+      cardList.prepend(cardElement);
+      closeModal(cardModal);
+      cardForm.reset();
+      disableButton();
```

Comment:
* There is an error when I add a new card https://snipboard.io/6Vx3hR.jpg 


The `disableButton` function requires 2 arguments: `buttonEl, config`. Please, pass the submit button and `settings` here


```js
      disableButton(submitBtn, settings);

```